pkg/infrastructure/gcp/clusterapi/clusterapi.go

```diff
@@ -0,0 +1,48 @@
+package clusterapi
+
+import (
+	"context"
+
+	"github.com/openshift/installer/pkg/infrastructure/clusterapi"
+	gcptypes "github.com/openshift/installer/pkg/types/gcp"
+)
+
+type Provider struct { // nolint:revive
+}
+
+var _ clusterapi.PreProvider = (*Provider)(nil)
+var _ clusterapi.IgnitionProvider = (*Provider)(nil)
+var _ clusterapi.InfraReadyProvider = (*Provider)(nil)
+var _ clusterapi.PostProvider = (*Provider)(nil)
+
+// Name returns the name for the platform.
+func (_ Provider) Name() string { // nolint:stylecheck
+	return gcptypes.Name
+}
+
+// PreProvision is called before provisioning using CAPI controllers has initiated.
+// GCP resources that are not created by CAPG (and are required for other stages of the install) are
+// created here using the gcp sdk.
+func (_ Provider) PreProvision(ctx context.Context, in clusterapi.PreProvisionInput) error { // nolint:stylecheck,revive
```

Comment:
```suggestion
func (Provider) PreProvision(ctx context.Context, in clusterapi.PreProvisionInput) error {
```
Sorry didnt realize the linter would complain. Lets not no lint. It seems like the error message is suggesting to fix by omitting the variable name